eva-stats/stats_accessioning.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import psycopg2
+import click
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query, execute_query
+from ebi_eva_common_pyutils.mongo_utils import get_mongo_connection_handle
+from ebi_eva_common_pyutils.config_utils import get_pg_metadata_uri_for_eva_profile, get_properties_from_xml_file
+from ebi_eva_common_pyutils.logger import logging_config
+
+
+logger = logging_config.get_logger(__name__)
+
+
+def get_handles(private_config_xml_file):
+    properties = get_properties_from_xml_file("production", private_config_xml_file)
+    mongo_hosts_and_ports = str(properties['eva.mongo.host'])
+    mongo_host, mongo_port = get_mongo_primary_host_and_port(mongo_hosts_and_ports)
+    mongo_username = str(properties['eva.mongo.user'])
+    mongo_password = str(properties['eva.mongo.passwd'])
+
+    mongo_handle = get_mongo_connection_handle(mongo_username, mongo_password, mongo_host)
+    metadata_handle = psycopg2.connect(get_pg_metadata_uri_for_eva_profile(
+        "development", private_config_xml_file), user="evadev")
+
+    return mongo_handle, metadata_handle
+
+
+def get_mongo_primary_host_and_port(mongo_hosts_and_ports):
+    """
+    :param mongo_hosts_and_ports: All host and ports stored in the private settings xml
+    :return: mongo primary host and port
+    """
+    for host_and_port in mongo_hosts_and_ports.split(','):
+        if '001' in host_and_port:
+            properties = host_and_port.split(':')
+            return properties[0], properties[1]
+
+
+def get_stats_from_accessioning_db(mongo_handle, metadata_handle, provided_assemblies):
+    logger.info("Getting counts for RS IDs (Clustered Variants)")
+    store_accessioning_counts(mongo_handle, metadata_handle, 'clusteredVariantEntity', provided_assemblies)
+    logger.info("Getting counts for SS IDs (Submitted Variants)")
+    store_accessioning_counts(mongo_handle, metadata_handle, 'submittedVariantEntity', provided_assemblies)
+
+
+def store_accessioning_counts(mongo_handle, metadata_handle, collection_name, provided_assemblies):
```

Comment:
Do you think it would be possible to grab the variant creation date (Year-Month) for each assembly during the aggregation pipeline. Looking at studies might be interesting as well, but it would only apply to SSid.

Reply:
It is possible with a different query but for large studies it has given a timeout error. We would need to get the created date from one of the variants in each study. Something like:
```
mongos> db.submittedVariantEntity.aggregate([{$match: { seq:"GCA_000001635.1"} }, {$group:{_id:"$study", example_accession:{$first:"$createdDate"} , count:{$sum:1} } } ] )
{ "_id" : "PRJEB38379", "example_accession" : ISODate("2020-05-21T14:52:14.519Z"), "count" : 100 }
{ "_id" : "PRJEB39892", "example_accession" : ISODate("2020-09-21T15:22:49.634Z"), "count" : 349 }

```